pkg/controller/servicemesh/controlplane/reconciler.go

```diff
@@ -116,7 +116,6 @@ func (r *controlPlaneInstanceReconciler) Reconcile(ctx context.Context) (result
 		// error handling
 		defer func() {
 			if err != nil {
-				r.renderings = nil
```

Comment:
Will the renderings be regenerated?

Reply:
Do they need to be? Wouldn't the regenerated renderings be identical? AFAICT the line I removed only caused the renderings to be regenerated when an error occurred during creation of the resources in these renderings. The next reconcile attempt generated the exact same renderings.

Of course, if a user updates the SMCP, its generation changes, the renderings are then set to nil here: https://github.com/Maistra/istio-operator/blob/53de3522d4b4297de7e6ba6b0b2740da2a211448/pkg/controller/servicemesh/controlplane/reconciler.go#L687

Reply:
Okay.  I thought I put that in for some reason, e.g. some other place checking to see if `renderings == nil`, but maybe that's what's causing the issue you're seeing (i.e. it was used to revert the component conditions).